src/ui/win/win_ui.cpp

```diff
@@ -66,6 +66,19 @@ namespace WindowsUi
                 SelectObject( hdc, hOldPen );
                 DeleteObject( hPen );
             }
+
+            for ( std::vector < UiWindowWin::EllipseToDraw >::const_iterator ellipse = _window->_ellipse.cbegin(); ellipse != _window->_ellipse.cend(); ++ellipse ) {
+                const int xTopLeft = static_cast<int>( ellipse->topLeft.x * xFactor);
+                const int yTopLeft = static_cast<int>( ellipse->topLeft.y * yFactor);
+                const int xBottomRight = static_cast<int>( ellipse->bottomRight.x * xFactor);
+                const int yBottomRight = static_cast<int>( ellipse->bottomRight.y * yFactor);
+
+                HPEN hPen = CreatePen( PS_SOLID, 1, RGB( ellipse->color.red, ellipse->color.green, ellipse->color.blue ) );
+                HGDIOBJ hOldPen = SelectObject( hdc, hPen );
+                Arc( hdc, xTopLeft, yTopLeft, xBottomRight, yBottomRight, 0, 0, 0, 0 );
```

Comment:
What if we use [Ellipse](https://docs.microsoft.com/en-us/windows/desktop/api/wingdi/nf-wingdi-ellipse) function instead? Then we just need to name variables as `left, right, top, bottom`.

Reply:
I've read the documentation. The naming of the function is misleading. But essentially, they are the same.
![image](https://user-images.githubusercontent.com/14083887/53001017-a91af180-3464-11e9-8b30-012b13de7e82.png)

Reply:
In my opinion we should use `Ellipse` function as it has less arguments and doesn't rise questions why we need to put so many zeros in function parameters :)

Regarding the naming: x coordinate of top left corner is left position, y coordinate of top left corner if top position. Naming a variable as `xTopLeft` is equal to `left` but the second case doesn't have excessive data in names. That's what I mean :) In most of image processing applications people use 4 sides instead of corners.

Reply:
So this refers to the sides. I see now.. Ok I'll be changing this in my next commit. 👍

Reply:
![image](https://user-images.githubusercontent.com/14083887/53002599-006e9100-3468-11e9-8edb-7714eafd2ec1.png)
I got this from using Ellipse.

Reply:
![image](https://user-images.githubusercontent.com/14083887/53002800-8a1e5e80-3468-11e9-8812-26248bf10314.png)
This is Arc.

Reply:
Then in this case you're totally right :)